src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -132,13 +133,15 @@ class Simplex_tree {
 
   struct Key_simplex_base_real {
     Key_simplex_base_real() : key_(-1) {}
+    Key_simplex_base_real(Simplex_key k) : key_(k) {}
     void assign_key(Simplex_key k) { key_ = k; }
     Simplex_key key() const { return key_; }
    private:
     Simplex_key key_;
   };
   struct Key_simplex_base_dummy {
     Key_simplex_base_dummy() {}
+    Key_simplex_base_dummy([[maybe_unused]] Simplex_key k) {}
```

Comment:
Your version is ok, but if you want, in general, I use the less verbose option of not naming the argument (except when naming it is needed for assertions, doc, etc)
```suggestion
    Key_simplex_base_dummy(Simplex_key) {}
```